### DISCUSSION TITLE
Any thoughts on building a standard contracts for `ReentrancyGuard` and `Timelock`

### DISCUSSION THREAD

**Discussion by timothyxbank** [OpenZeppelin/cairo-contracts#220] [Design]
These 2 contracts are considered to be very important on L1 contracts.
For Timelock, Most of the contract replace the owner with `Timelock` to prevent unnecessary privileged on the account. 
For ReentrancyGuard, we can use check-effect-interaction as a convention to prevent this kind of attack, but it should be better to have a solid library to prevent this issue for us, similar to `Ownable`.

Any thoughs on implementing these 2 contracts?

**Comment by franceccon** [comment]
Been thinking about reentrancy too. How would it look like in Cairo? I think something like `ReentrancyGuard_start()` and `ReentrancyGuard_end()`.

**Reply by timothyxbank** [reply]
yeah, I think your version is good and understandable

**Reply by martriay** [reply]
I hate it but I agree, that's the best. I'll create an issue.

**Reply by timothyxbank** [reply]
than's nice, thanks @martriay, and how about `Timelock` then, any thoughts?

**Reply by martriay** [reply]
> than's nice, thanks @martriay, and how about `Timelock` then, any thoughts?

There was an open issue for that already :)
